sdk/compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -715,6 +732,14 @@ generateSrcFromLf env = noLoc mod
         && (LF.moduleNameString lfModName /= "GHC.Prim")
         && not (LF.unExprValName lfName `Set.member` classMethodNames)
 
+    shouldExposeExport :: LFC.QualName -> Bool
```

Comment:
Please explain why these 3 cases in a comment.

Reply:
Turns out this additional check wasn't needed, as it simply made assertions about stdlib/prim (not needed as they export everything), and about typeclasses methods, which are exported with the typeclass def, and would thus be caught by the `qualNameToSynName name `MS.member` classReexportMap` clause in `allExports`. I've removed it in the split daml-script PR